libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/MapboxHybridSpeechPlayer.kt

```diff
@@ -0,0 +1,100 @@
+package com.mapbox.navigation.ui.voice.api
+
+import android.content.Context
+import androidx.annotation.UiThread
+import com.mapbox.navigation.ui.base.api.voice.SpeechPlayer
+import com.mapbox.navigation.ui.base.api.voice.SpeechPlayerCallback
+import com.mapbox.navigation.ui.base.model.voice.Announcement
+import com.mapbox.navigation.ui.base.model.voice.SpeechState
+import java.util.LinkedList
+import java.util.Queue
+import java.util.concurrent.ConcurrentLinkedQueue
+
+/**
+ * Hybrid implementation of [SpeechPlayer] combining [MapboxOnboardSpeechPlayer] and
+ * [MapboxOffboardSpeechPlayer] speech players.
+ * @property context Context
+ * @property accessToken String
+ * @property language [Locale] language (ISO 639)
+ */
+@UiThread
+class MapboxHybridSpeechPlayer(
+    private val context: Context,
+    private val accessToken: String,
+    private val language: String
+) : SpeechPlayer {
+
+    private val queue: Queue<SpeechState.ReadyToPlay> = ConcurrentLinkedQueue()
+    private val offboardSpeechPlayer: MapboxOffboardSpeechPlayer =
+        MapboxOffboardSpeechPlayer(context, accessToken, language)
+    private val onboardSpeechPlayer: MapboxOnboardSpeechPlayer =
+        MapboxOnboardSpeechPlayer(context, language)
+    private val clientCallbacks: LinkedList<SpeechPlayerCallback> = LinkedList()
+    private var localCallback: SpeechPlayerCallback = object : SpeechPlayerCallback {
+        override fun onDone(state: SpeechState.DonePlaying) {
+            val announcement = queue.poll()
+            val clientCallback = clientCallbacks.first
```

Comment:
Are we only calling onDone for the first callback? Seems like it should remove from `clientCallbacks` 🤔

Reply:
Great catch! While working with the queue forgot that `first` doesn't remove from the list 😅

Reply:
Will replace by `removeFirst` 👍

Reply:
Actually, I'm going to replace it by `ConcurrentLinkedQueue` (and will use `poll`) to avoid any potential races.